DEHPEcosimPro/ViewModel/Rows/ParameterDifferenceViewModel.cs

```diff
@@ -0,0 +1,324 @@
+                }
+            }
+        }
+
+        /// <summary>
+        /// Compute values and add it to the object
+        /// </summary>
+        /// <param name="index">number of iteration</param>
+        /// <param name="isOptions">Has the Thing Options?</param>
+        /// <param name="isState">Has the Thing States?</param>
+        /// <returns><see cref="ParameterDifferenceRowViewModel"/></returns>
+        private ParameterDifferenceRowViewModel PopulateParameterDifferenceRowViewModel(int index, bool isOptions, bool isState)
+        {
+            var setOfNewValues = this.listofsetOfNewValues[index].ActualValue;
+            var setOfOldValues = this.listofsetOfOldValues[index].ActualValue;
+
+            object NewValue;
+            object OldValue;
+            object Name = this.ModelCode();
+
+            if (setOfNewValues.Count > 1)
+            {
+                NewValue = setOfNewValues.ToString();
+            }
+            else
+            {
+                NewValue = setOfNewValues.FirstOrDefault();
+            }
+
+            if (setOfOldValues.Count > 1)
+            {
+                OldValue = setOfOldValues.ToString();
+            }
+            else
+            {
+                OldValue = setOfOldValues.FirstOrDefault();
+            }
+
+
+            if (isOptions && isState)
+            {
+                Name = Name + $"\\{this.listofsetOfNewValues[index].ActualOption.ShortName}\\{this.listofsetOfNewValues[index].ActualState.ShortName}";
+            }
+            else if (isOptions && !isState)
+            {
+                Name = Name + $"\\{this.listofsetOfNewValues[index].ActualOption.ShortName}";
+            }
+            else if (!isOptions && isState)
+            {
+                Name = Name + $"{this.listofsetOfNewValues[index].ActualState.ShortName}";
+            }
+            else if (!isOptions && !isState)
+            {
+                Name = this.ModelCode();
+            }
+
+            this.CalculateDiff(OldValue, NewValue, out string Difference, out string PercentDiff);
+
+            return new ParameterDifferenceRowViewModel(this.OldThing, this.NewThing, Name, OldValue, NewValue, Difference, PercentDiff);
+
+        }
+
+        /// <summary>
+        /// Add The valueSet to list
+        /// </summary>
+        /// <param name="option"><see cref="Option"/></param>
+        /// <param name="state"><see cref="ActualFiniteState"/></param>
+        private void PopulateListOfSets(Option option, ActualFiniteState state)
+        {
+            try
+            {
+                this.listofsetOfOldValues.Add(this.OldThing.QueryParameterBaseValueSet(option, state));
+                this.listofsetOfNewValues.Add(this.NewThing.QueryParameterBaseValueSet(option, state));
+            }
+            catch (Exception e)
+            {
+                Console.WriteLine(e);
+                throw;
```

Comment:
Does it need to throw an exception here? If so, please add a log entry before re-thowing, and throw a more specific exception e.g. ArgumentNullException with a message